### PR TITLE
Preview implementation of AzureCliCredential

### PR DESCRIPTION
**Pull request by chlowell**
This credential gets tokens by subprocessing the Azure CLI's `get-access-token` command, and is part of `DefaultAzureCredential`.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Release History
 
-## 1.3.1 (Unreleased)
+## 1.4.0b1 (Unreleased)
+- Added `AzureCliCredential`, which authenticates with the identity logged in
+to the Azure CLI. This credential is part of `DefaultAzureCredential` by
```
Comment (jongio) [inline sdk/identity/azure-identity/CHANGELOG.md]: Please sync with rest of lang devs to ensure we are on the same page across langs on whether or not DAC has CLICred by default.

I highly prefer it to be enabled by default so developer doesn't have to do anything to configure it.

File: sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    """Invoke 'az' from a directory on $PATH or a well-known install location, not the executing program's directory"""
+
+    path = os.environ.get("PATH")
+
+    if sys.platform.startswith("win"):
+        if path:
+            return path.split(";")[0]
+
+        # no system path; check well-known install locations
+        cli_path = "Microsoft SDKs\\Azure\\CLI2\\wbin"
```
Comment (lmazuel) [inline sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py]: So if I pip install the CLI on Windows, I can't use this class?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py]: We only land here when the environment has no path defined. So if pip places the CLI outside the environment's path and these Program Files locations, that's correct, you can't use this class.

Comment (lmazuel) [reply sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py]: discussed offline got my answer

File: sdk/identity/azure-identity/azure/identity/__init__.py [hunk 3]
```diff
@@ -23,6 +24,7 @@
 
 __all__ = [
     "AuthorizationCodeCredential",
+    "AzureCliCredential",
```
Comment (schaabs) [inline sdk/identity/azure-identity/azure/identity/__init__.py]: I thought we decided to to export this type. No?

**Pull request merged**